### PR TITLE
Multi account guidance for EKS clusters

### PR DESCRIPTION
**Pull request by liwadman**
*Issue #, if available:*

*Description of changes:*

Added new document to security best practices, informing customers about different patterns of multi account strategies, why they should chose a multi account strategy, and information on the pros and cons of either strategy.

authored by liwadman@/asramaa@ , please reach out for any Qs or revisions.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jicowan — COMMENTED** [review]
Apologies for the delay. I will review this PR this weekend.

File: content/security/docs/multiaccount.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Workload accounts can then be further broken down by software development lifecycle or other requirements if appropriate. For example a given workload can have a production account, a development account, or accounts for hosting instances of that workload in a specific region. [More information](https://docs.aws.amazon.com/whitepapers/latest/organizing-your-aws-environment/organizing-workload-oriented-ous.html) is available in this AWS whitepaper.
+
+You can adopt the following approaches when implementing EKS Multi account strategy:
+
+## Centralized EKS Cluster
+
+In this approach, your EKS Cluster will be deployed in a single AWS account called the `Cluster Account`. Using [IAM roles for Service Accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) and [AWS Resource Access Manager (RAM)](https://aws.amazon.com/ram/) you can adapt a multi account strategy for your multi tenant EKS cluster. The cluster account will contain the VPC, subnets, EKS cluster, EC2/Fargate compute resources, and any additional networking or configurations needed to run your EKS cluster.
```
Comment (jicowan) [inline content/security/docs/multiaccount.md:24]: How will RAM be used in this proposed design?

File: content/security/docs/multiaccount.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+In a multi workload account strategy for multi tenant cluster, AWS accounts typically align with [kubernetes namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) as a mechanism for isolating groups of resources. [Best practices for tenant isolation](/security/docs/multitenancy/) within an EKS cluster should still be followed when implementing a multi account strategy for multi tenant EKS clusters.
+
+It is possible to have multiple `cluster accounts` in your AWS organization, and it is a best practice to have multiple `cluster accounts` that align with your software development lifecycle needs. For workloads operating at a very large scale, you may require multiple `cluster accounts` to ensure that there are enough kubernetes and AWS service quotas available to all your workloads.
+
+| ![](./images/multi-account-eks.jpg) |
+|:--:|
```
Comment (jicowan) [inline content/security/docs/multiaccount.md:31]: We are planning to remove references to this from the best practices guide and our documentation as the service team does not test for it and it isn't guaranteed to work.

File: content/security/docs/multiaccount.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+AWS API resources and other Resources that use IAM authentication in a workload account can only be accessed by credentials for IAM roles in that same workload account, except where cross account access is capable and has been explicity enabled.
+
+#### Enabling IRSA for cross account access
+
+To enable IRSA for workloads in your Cluster Account to access resources in your Workload accounts, you first must create an IAM OIDC identity provider in your workload account. This can be done with the same procedure for setting up IRSA, except the Identity Provider will be created in the workload account: https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html
```
Comment (jicowan) [inline content/security/docs/multiaccount.md:55]: This is not the only way to assume a role in another account. A service account can be mapped to an IAM role that allows it to assume a role in another account.

**Review by jicowan — COMMENTED** [review]
@liwadman Thanks for the PR. I'd like you to work with @sheetaljoshi and @LukeMwila on this. They were tasked with creating a section on 1 vs. many clusters.

File: content/security/docs/multiaccount.md (hunk 3 above)
Comment (liwadman) [reply content/security/docs/multiaccount.md:55]: Yeah bad that is not scaleable, it puts pressure on the amount of available IAM roles in the cluster account and customers will quickly run up on this issue.

Comment (liwadman) [reply content/security/docs/multiaccount.md:55]: After working with many customers hitting this problem of scale, IRSA is a much more practical solution.

File: content/security/docs/multiaccount.md (hunk 2 above)
Comment (jicowan) [reply content/security/docs/multiaccount.md:31]: I meant that you shouldn't try to join nodes from different accounts to a cluster in a different account. The team doesn't test that use case.

File: content/security/docs/multiaccount.md (hunk 3 above)
Comment (jicowan) [reply content/security/docs/multiaccount.md:55]: It's still using IRSA but the role the pod assumes allows it to assume a role in another account.

File: content/security/docs/multiaccount.md (hunk 2 above)
Comment (liwadman) [reply content/security/docs/multiaccount.md:31]: ack on that - yeah nodes belonging to other accounts isn't a part of our pattern, and I wouldn't recommend it. Can add clarifying notes into the doc that nodes belong to the cluster account.

File: content/security/docs/multiaccount.md (hunk 3 above)
Comment (liwadman) [reply content/security/docs/multiaccount.md:55]: Role chaining means that the app authors need to bring this into their code or config to account for, and it means in a decently busy AWS account they'll quickly run against the default IAM role per account quota.

It also means that customer code has access to a role session in a customer account, which of course has no default entitlements but it means their code could potentially access resources in that account, which is against the isolation objective we're trying to achieve.

**Pull request merged**